packages/governance/src/votes/votes.cairo

```diff
@@ -180,21 +174,54 @@ pub mod ERC20VotesComponent {
     // Internal
     //
 
+    impl ERC721VotesImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        impl ERC721: ERC721Component::HasComponent<TContractState>,
+        +ERC721Component::ERC721HooksTrait<TContractState>,
+        +Drop<TContractState>
+    > of TokenVotesTrait<ComponentState<TContractState>> {
+        /// Returns the number of voting units for a given account.
+        ///
+        /// This implementation is specific to ERC721 tokens, where each token
+        /// represents one voting unit. The function returns the balance of
+        /// ERC721 tokens for the specified account.
+        fn get_voting_units(
+            self: @ComponentState<TContractState>, account: ContractAddress
+        ) -> u256 {
+            let erc721_component = get_dep_component!(self, ERC721);
+            erc721_component.balance_of(account).into()
+        }
+    }
+
+    impl ERC20VotesImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC20: ERC20Component::HasComponent<TContractState>,
+        +ERC20Component::ERC20HooksTrait<TContractState>
+    > of TokenVotesTrait<ComponentState<TContractState>> {
+        /// Returns the number of voting units for a given account.
+        ///
+        /// This implementation is specific to ERC20 tokens, where the balance
+        /// of tokens directly represents the number of voting units.
+        fn get_voting_units(
+            self: @ComponentState<TContractState>, account: ContractAddress
+        ) -> u256 {
+            let erc20_component = get_dep_component!(self, ERC20);
+            erc20_component.balance_of(account)
+        }
+    }
+
     #[generate_trait]
     pub impl InternalImpl<
         TContractState,
         +HasComponent<TContractState>,
-        impl ERC20: ERC20Component::HasComponent<TContractState>,
-        +ERC20Component::ERC20HooksTrait<TContractState>,
+        impl TokenTrait: TokenVotesTrait<ComponentState<TContractState>>,
```

Comment:
```suggestion
       +TokenVotesTrait<ComponentState<TContractState>>,
```

We don't need to explicitly call the implementation.

Reply:
Better this way, thanks! I still get confused about trait bounds and `impl` inside the generics.